mllib/src/test/scala/org/apache/spark/ml/tree/impl/TreeTests.scala

```diff
@@ -124,8 +129,8 @@ private[ml] object TreeTests extends SparkFunSuite {
    *       make mistakes such as creating loops of Nodes.
    */
   private def checkEqual(a: Node, b: Node): Unit = {
-    assert(a.prediction === b.prediction)
-    assert(a.impurity === b.impurity)
+    assert(a.prediction ~== b.prediction absTol 1e-8)
+    assert(a.impurity ~== b.impurity absTol 1e-8)
```

Comment:
it's strange that we are using different tolerances everywhere as opposed to a few specific ones (say 1e-7 for absolute and 1e-3 for relative).  But since they are different everywhere, you are right, it would be overengineering, so I think it's ok to leave as is.